### PR TITLE
Add PresignRequest API

### PR DESCRIPTION
**Pull request by EngHabu**
Expose PreSignRequest for storage clients

Signed-off-by: Haytham Abuelfutuh <haytham@afutuh.com>

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue

https://github.com/flyteorg/flyte/issues/2263

### CODE REVIEW

File: storage/stow_store_test.go [hunk 1]
```diff
@@ -143,6 +149,81 @@ func TestAwsBucketIsNotFound(t *testing.T) {
...
+			CreateContainerCb: func(name string) (stow.Container, error) {
+				if name == container {
+					return newMockStowContainer(container), nil
+				}
+				return nil, fmt.Errorf("container is not supported")
+			},
+		}, false, testScope)
+		assert.NoError(t, err)
+
+		actual, err := s.CreateSignedURL(context.TODO(), DataReference("https://container/path"), SignedURLProperties{})
```
Comment (wild-endeavor) [inline storage/stow_store_test.go:175]: i'm confused why this happy path starts with https://... shouldn't this be s3 or gcs?

**Comment by katrogan** [comment]
oh this is great! we should move admin to use this for signed urls too once this is in

File: storage/stow_store_test.go (hunk 1 above)
Comment (EngHabu) [reply storage/stow_store_test.go:175]: Any valid url format is ok.. no?

**Review by katrogan — DISMISSED** [review]

**Review by katrogan — DISMISSED** [review]

File: storage/stow_store_test.go (hunk 1 above)
Comment (wild-endeavor) [reply storage/stow_store_test.go:175]: i guess, i just find it confusing.  would this ever be called like this?

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**